File: Array/sorting/insertionsort.cpp
// bubble sort
#include <vector>
#include  <stdio.h> 
#include <iostream>
using namespace std;

void insertionsort(vector<int>& arr, int n)
{
    for (int i = 1; i<n; i++){
        int j = i-1;
        int temp = arr[i];
        while (j>=0){
            if (arr[j] > temp)
            {
                arr[j+1] = arr[j];

            }
            else{
                break;
            }
            j -=1;
        }
        arr[j+1] = temp;
    }
}
int main(){

    vector<int> arr = {6,9,4,1,5,4};
    int n = 6;
    insertionsort(arr, n);

    for (int i=0; i < n; i++){
        cout<<arr[i]<<endl;
    }
}

File: Array/sorting/insertionsort.py

def insertionsort(arr, n):
    for indx, i in enumerate(range(1,n)):
        temp = arr[i]
        j = i-1
        while j >= 0:
            if (arr[j] > temp):
                arr[j+1] = arr[j]
            else:
                break
            j -=1
        
        arr[j+1] = temp

    return arr

if __name__ == "__main__":
    arr = [10,3,7,4,5,9,2]
    n = 7

    sorted_arr = insertionsort(arr, n)

    print(sorted_arr)
